src/common/commonutils/FileUtils.c

```diff
@@ -1199,6 +1198,16 @@ int CheckTextIsNotFoundInFile(const char* fileName, const char* text, char** rea
     return result;
 }
 
+static bool IsValidGrepArgument(const char* text)
+{
+    return IsValidDaemonName(text);
+}
+
+static bool IsValidCommentCharacter(char c)
+{
+    return ((0 == isalnum(c)) && ('#' != c)) ? false : true;
+}
+
 int CheckMarkedTextNotFoundInFile(const char* fileName, const char* text, const char* marker, char commentCharacter, char** reason, void* log)
 {
     const char* commandTemplate = "grep -v '^%c' %s | grep %s";

```

Comment:
Do we need to call grep twice here? The first one just tests the first character in line, the latter is seeking for a text which is not a regex, so this could be achieved by reading a file line by line and calling strstr() on these if I'm correct. We could save the need for 'text' and 'filename' arguments validation, forking a process and relying on timeouts. If you believe this optimisation is worth considering, please do in a separate PR so it's easier to review.

Reply:
This sounds like an interesting idea (thanks!) however it's nothing that we should block or delay release for now if you follow what I am saying. Thx